WalletWasabi.Packager/Program.cs

```diff
@@ -503,18 +510,72 @@ private static void Publish()
 
 					StartProcessAndWaitForExit("wsl", BinDistDirectory, arguments: arguments);
 
-					IoHelpers.TryDeleteDirectoryAsync(debFolderPath).GetAwaiter().GetResult();
+					await IoHelpers.TryDeleteDirectoryAsync(debFolderPath).ConfigureAwait(false);
 
 					string oldDeb = Path.Combine(BinDistDirectory, $"{ExecutableName}_{VersionPrefix}_amd64.deb");
 					string newDeb = Path.Combine(BinDistDirectory, $"Wasabi-{VersionPrefix}.deb");
 					File.Move(oldDeb, newDeb);
 
-					IoHelpers.TryDeleteDirectoryAsync(publishedFolder).GetAwaiter().GetResult();
+					await IoHelpers.TryDeleteDirectoryAsync(publishedFolder).ConfigureAwait(false);
 					Console.WriteLine($"Deleted {publishedFolder}");
 				}
 			}
 		}
 
+		/// <summary>Checks whether there are uncommitted changes.</summary>
+		/// <remarks>This is important to really release a build that corresponds with a git hash.</remarks>
+		private static void CheckUncommittedGitChanges()
+		{
+			string? gitStatus = StartProcessAndWaitForExit("git", workingDirectory: DesktopProjectDirectory, arguments: "status --porcelain", redirectStandardOutput: true)?.Trim();
+
+			if (!string.IsNullOrEmpty(gitStatus))
+			{
+				Console.WriteLine("BEWARE: There are uncommitted changes in the repository. Do you want to continue? (Y/N)");
+				ConsoleKeyInfo consoleKeyInfo = Console.ReadKey();
+
+				if (consoleKeyInfo.Key.ToString().ToLowerInvariant() != "y")
+				{
+					Console.WriteLine("\nExiting.");
+					Environment.Exit(1);
+				}
+			}
+		}
+
+		/// <summary>
+		/// Gets information about .NET SDK and .NET runtime so that deterministic build is easier to reproduce.
+		/// </summary>
+		/// <returns>Array of lines to write to a <c>BUILDINFO.txt</c> file.</returns>
+		private static string[] GetBuildInfoData()
+		{
+			// .NET runtime version. We rely on the fact that this version is the same as if we run "dotnet" command. This should be a safe assumption.
+			Version runtimeVersion = Environment.Version;
+
+			// Get .NET SDK version.
+			string? sdkVersion = StartProcessAndWaitForExit("dotnet", workingDirectory: DesktopProjectDirectory, arguments: "--version", redirectStandardOutput: true)?.Trim();
+
+			if (sdkVersion is null)
+			{
+				throw new InvalidOperationException($"Failed to get .NET SDK version.");
+			}
+
+			// Get git commit ID.
+			string? gitCommitId = StartProcessAndWaitForExit("git", workingDirectory: DesktopProjectDirectory, arguments: "rev-parse HEAD", redirectStandardOutput: true)?.Trim();
+
+			if (gitCommitId is null)
+			{
+				throw new InvalidOperationException($"Failed to get git commit ID.");
+			}
+
+			string[] lines = new string[]
+			{
+				$".NET Runtime: {runtimeVersion}",
+				$".NET SDK:     {sdkVersion}",
+				$"git hash:     {gitCommitId}",
```

Comment:
Nitpick:
```suggestion
				$"Git Hash:     {gitCommitId}",
```
Or `Git Commit Hash: ...` ?

Reply:
Tx, modified